qCC/ccCommandLineCommands.cpp

```diff
@@ -2743,6 +2744,29 @@ static std::pair<ScalarType, ScalarType> GetSFRange(const CCCoreLib::ScalarField
 	return { thisMinVal, thisMaxVal };
 }
 
+static ScalarType GetSFValue(ccPointCloud* pc, int sfIndex, ScalarType value, USE_SPECIAL_SF_VALUE useVal)
+{
+	//should be handled way before this point this is just safety
+	if (pc)
```

Comment:
another option is to use a reference to a cloud instead of a pointer

Reply:
so if i define like the folowwing the pc check is not necessary?

static ScalarType GetSFValue(ccPointCloud pc, int sfIndex, ScalarType value, USE_SPECIAL_SF_VALUE useVal)

and i call it with dereferenced value?

&desc.pc

Reply:
Yes, but it should be static ScalarType GetSFValue(const ccPointCloud& pc, int sfIndex, ScalarType value, USE_SPECIAL_SF_VALUE useVal)

(to avoid a copy of the cloud)